utilities/include/vec4.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef VEC4_HPP
+#define VEC4_HPP
+
+#include <cstddef>
```

Comment:
A comment here stating what we need this include for would be nice. Something like `// for the std::size_t datatype`.